core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -35,8 +36,24 @@ private[spark] class AppStatusStore(
     val store: KVStore,
     val listener: Option[AppStatusListener] = None) {
 
+  /**
+   * This method contains an automatic retry logic and tries to get a valid [[v1.ApplicationInfo]].
```

Comment:
nit: shall we use `` instead of  [[]]? @HyukjinKwon

Reply:
`ApplicationInfo` seems a proper class which can properly converted into Java class via `Unidoc` in principal. If it works, let's uses `[[...]]`. If it fails to generate the documentation, let's stick to `` `...` ``.

Reply:
We should use `` `...` `` in case the classes are a trait or a type alias which doesn't generate a canonical Java class.